R/adjust_household_contact_matrix.R

```diff
@@ -0,0 +1,50 @@
+#' @title Adjust Household Contact Matrix
+#' 
+#' This function is used internally within [predict_setting_contacts()]
+#'
+#' @param setting_matrices setting matrix
+#' @param population population information
+#' @param household_size single number of the size of a household
+#' @param household_contact_rate default of 1
+#' @param model_mean_other_householders default of 2.248971
+#'
+#' @return contact matrix that has household adjustment
+#' @author Nick Golding
+#' @export
+adjust_household_contact_matrix <- function(setting_matrices,
+                                            population,
+                                            household_size,
+                                            household_contact_rate = 1,
+                                            model_mean_other_householders = 2.248971) {
+  
+  # get the mean number of other household members from thpolymod data used to
+  # train the contact model
+  
+  # model_mean_other_householders <- socialmixr::polymod$participants %>%
```

Comment:
perhaps this snippet of code could be wrapped up in a function:
```
get_polymod_percapita_household_size <- function() {
  mean(socialmixr::polymod$participants$hh_size)
}
```
or just a data object, and passed in as the default argument:
```
model_percapita_household_size = get_polymod_percapita_household_size()
```

Reply:
that's because this is currently hard-coded for polymod data, though we will want to let users pass in the value for their own survey, via the `predict_setting_contacts()` function

Reply:
Ah I see how it used later on in the thread

Reply:
I assume you mean

`mean(socialmixr::polymod$participants$hh_size - 1)`

Otherwise we get `3.248971`

Reply:
No I don't, the minus 1 will now be done later, in computing `ratio`.
We want it without the minus 1 here so that it is easier for the users to understand what it means.

Reply:
Ah my comments weren't in sync - sorry I didn't pick up on that - I've removed the -1 part, since as you said, that happens later in the chain.